apps/api/src/domain/models/photograph/entities/PhotographDimensions.ts

```diff
@@ -0,0 +1,30 @@
+import { InternalError } from 'apps/api/src/lib/errors/InternalError';
+import { PartialDTO } from 'apps/api/src/types/partial-dto';
+import BaseDomainModel from '../../BaseDomainModel';
+
+// This is a value-object
+export default class PhotographDimensions extends BaseDomainModel {
+    // Do we need the unit in the prop name? Do we need other units?
+    readonly widthPX: number;
+
+    readonly heightPX: number;
+
+    constructor({ widthPX, heightPX }: PartialDTO<PhotographDimensions>) {
+        super();
+
+        this.widthPX = widthPX;
+
+        this.heightPX = heightPX;
+    }
+
+    rescale(scaleFactor: number): PhotographDimensions {
+        if (scaleFactor <= 0) {
```

Comment:
could we use (scaleFactor < 1) or is there cases where the px is less than 1 but greater than 0?

Reply:
You should be able to scale right down to 0X0 (but not including) or anything bigger. But there may be subtleties around this we'll need to capture in our next round of modelling.